sqlglot/dialects/snowflake.py

```diff
@@ -874,7 +874,12 @@ def _parse_location_path(self) -> exp.Var:
             ):
                 parts.append(self._advance_any(ignore_reserved=True))
 
-            return exp.var("".join(part.text for part in parts if part))
+            part_texts = (
+                f'"{part.text}"' if part.token_type == TokenType.IDENTIFIER else part.text
+                for part in parts
+                if part
+            )
+            return exp.Literal(this="".join(part_texts), is_string=False)
```

Comment:
This doesn't look right, a `Literal` with `is_string` set to false is a number. In my previous reply I meant that I'd prefer canonicalizing all URIs for staged paths as strings. This is a breaking change, but should be ok given that [Snowflake doesn't mind single quotes](https://github.com/tobymao/sqlglot/pull/4906#discussion_r2009086561).

Additionally, I think it should be possible to simplify the approach here by leveraging `_find_sql`, instead of creating a list of all the tokens that make up the URI path. So, we'd consume all tokens as we do now, but at the end we'd do `_find_sql(start, end)` and then use that text to build the string literal.

Reply:
Thanks for clarifying @georgesittas - updated the PR based on your suggestions. 👍 Had to adjust a couple of the existing tests to have `write_sql=...` represent the new output format with quotes.

Regarding the change with `_find_sql(..)` - I've noticed that this call also seems to be consuming the quotes `'` for quoted stage references, hence I had to add a switch that strips of any `'` at the beginning/end of the result string. 

See here: https://github.com/tobymao/sqlglot/pull/4906/files#diff-c3dbaf62f01fc7a9baf342a000c2351ebea03abdede048f92d234c44e11c7bfeR879-R880 I can imaging that this is most certainly not the best way to do this, I'm sure there's a more elegant way to achieve this :) - again, happy for any guidance you can give! 🙌